Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine+State.swift

```diff
@@ -70,15 +76,31 @@ extension ExperienceStateMachine {
                 if markComplete {
                     sideEffect = .processActions(experience.postExperienceActionFactory)
                 }
-                return Transition(toState: .idling, sideEffect: sideEffect)
+                return Transition(toState: .idling, sideEffect: sideEffect, resetObservers: true)
+            case let (.failing(targetState, retryEffect), .retry):
+                return Transition(toState: targetState, sideEffect: retryEffect)
+            case (.failing, .startExperience):
+                // do not reset observers here, as we want to maintain the attached analytics listener
+                // through the transition to idling and on to the new experience start attempt
+                return Transition(toState: .idling, sideEffect: .continuation(action), resetObservers: false)
```

Comment:
note: `resetObservers` is false here, as we are moving right on to starting a new experience and want to maintain those, I think? (the analytics observer namely)

Reply:
thats a good edge case catch, yes, it would. Need to think about how to handle that one. We don't have that issue on Android, as the check for published flow happens in a different spot inside the observer itself. But here, we'll need something to handle it.

Reply:
left this code as-is and updated in https://github.com/appcues/appcues-ios-sdk/pull/474/commits/57534732108e1c72635d662faae022491d6835d5 to always attach an analytics observer, but have it filter internally to avoid reporting analytics for any unpublished flow content (matching Android behavior)

Reply:
Perfect!